subprojects/gradle/build-metrics/src/main/kotlin/com/avito/android/plugin/build_metrics/CollectTeamcityMetricsTask.kt

```diff
@@ -0,0 +1,29 @@
+package com.avito.android.plugin.build_metrics
+
+import com.avito.android.stats.statsdConfig
+import com.avito.teamcity.teamcityCredentials
+import com.avito.utils.logging.ciLogger
+import org.gradle.api.DefaultTask
+import org.gradle.api.tasks.TaskAction
+import org.gradle.workers.WorkerExecutor
+import javax.inject.Inject
+
+abstract class CollectTeamcityMetricsTask @Inject constructor(
+    private val workerExecutor: WorkerExecutor
+) : DefaultTask() {
+
+    @TaskAction
+    fun action() {
+        val buildId: String? = System.getenv("TMCT_METRICS_BUILD_ID")
```

Comment:
I didn't use gradle properties because we can't have a default value.

Reply:
@Optional?

Reply:
`@Optional`

Reply:
https://github.com/avito-tech/avito-android/pull/213#discussion_r385032657